Attempt to use ripper-tags, if installed

sj26 opened the pull request:
[ripper-tags](https://github.com/tmm1/ripper-tags) providers much more comprehensive tags than [exuberant] ctags. For example, trying to find a tag for `Gem::Specification` in the vendored rubygems doesn't find the primary definition:

```
  # pri kind tag               file
> 1 F   c    Specification     /usr/local/rbenv/versions/2.7.1/lib/ruby/site_ruby/2.7.0/bundler/rubygems_ext.rb
               class:Gem
               class Specification
  2 F   c    Specification     /usr/local/rbenv/versions/2.7.1/lib/ruby/2.7.0/bundler/rubygems_ext.rb
               class:Gem
               class Specification
```

After using ripper-tags, the primary file is included (even if still not the default):
```
  # pri kind tag               file
> 1 F   c    Specification     /usr/local/rbenv/versions/2.7.1/lib/ruby/site_ruby/2.7.0/bundler/rubygems_ext.rb
               class:Gem
               class Specification
  2 F   c    Specification     /usr/local/rbenv/versions/2.7.1/lib/ruby/site_ruby/2.7.0/rubygems/resolver/specification.rb
               class:Gem.Resolver
               class Gem::Resolver::Specification
  3 F   c    Specification     /usr/local/rbenv/versions/2.7.1/lib/ruby/site_ruby/2.7.0/rubygems/specification.rb
               class:Gem inherits:Gem.BasicSpecification
               class Gem::Specification < Gem::BasicSpecification
  4 F   c    Specification     /usr/local/rbenv/versions/2.7.1/lib/ruby/2.7.0/bundler/rubygems_ext.rb
               class:Gem
               class Specification
  5 F   c    Specification     /usr/local/rbenv/versions/2.7.1/lib/ruby/2.7.0/rubygems/resolver/specification.rb
               class:Gem.Resolver
               class Gem::Resolver::Specification
  6 F   c    Specification     /usr/local/rbenv/versions/2.7.1/lib/ruby/2.7.0/rubygems/specification.rb
               class:Gem inherits:Gem.BasicSpecification
               class Gem::Specification < Gem::BasicSpecification
```

Personally I don't mind that this requires running `rbenv ctags` again after a `gem install ripper-tags` — the extra step is worth it for me. But I'll probably combine this with [rbenv-default-gems](https://github.com/rbenv/rbenv-default-gems) somehow, if I can figure out ordering.

Fixes #4.

sj26 committed:
Attempt to use ripper-tags, if installed

tpope commented:
To start, see tpope/gem-ctags#10, a similar feature request which I ultimately addressed by enabling the use of [a simple wrapper script](https://github.com/tpope/gem-ctags/issues/10#issuecomment-304421664). That wrapper script will work with rbenv-ctags, too, and I am inclined to recommend it as a first pass solution.

I am not opposed to an additional higher level solution, but I don't think `command -v ripper-tags` is gonna cut it. That will return true if a shim exists, but when we actually try to call it, there's no guarantee the gem will actually be available, meaning you could get an error like

```
rbenv: ripper-tags: command not found

The `ripper-tags' command exists in these Ruby versions:
  2.5.0
```

Ideally, we'd run _after_ rbenv-default-gems, and then use `ripper-tags` if and only if rbenv-default-gems installed it. But short of renaming the plugin, is there a good way to guarantee that run order?

sj26 committed:
Look for ripper-tags in this rbenv version

Avoids:

   rbenv: ripper-tags: command not found

   The `ripper-tags' command exists in these Ruby versions:
     2.5.0

.. with the caveat that it will only work with a ripper-tags installed
into an rbenv ruby, and not at a system level etc -- although unlikely,
and this rbenv plugin should be about conventional rbenv usage.

sj26 commented:
What about using rbenv-which? That avoids the shim-exists-but-gem-uninstalled problem.

I like the idea of a wrapper script — it's a neat solution! — but I don't like the maintenance which comes with it. I'd have to subvert homebrew's default installed exuberant ctags binary, or install it unlinked, and then install and maintain a homebrew-aware but not homebrew managed wrapper script somehow. I'd much prefer a ripper tags aware integration, if possible, even if imperfect.

How would you feel about this code path as an optional, best-effort optimisation?

tpope commented:
`rbenv-which` sounds good. But I think we need to back up a step and add a `--[no-]ripper` command line option. You shouldn't need to `gem install` and `gem remove` just to toggle between the two. And I'm torn on whether `gem ctags` should do the automatic detection, or just the after install hook.

I'm actually not sure what the best way to add an option parser to a bash script is. Normally I use `getopts` which is easy enough but only supports single letter options. I'm totally fine with `-r`/`-R` if that's what it takes to ship it.

>  I'd have to subvert homebrew's default installed exuberant ctags binary

Just put the custom script earlier in your path. Personally I use `~/.local/bin`. `~/.rbenv/bin` would be fitting if you don't mind an untracked file.

tpope commented:
I added a simple option parser. It should be straightforward to add a `--ripper` option.

sj26 commented:
Closing in favour of #15
